File: syncloud/src/main/java/org/syncloud/android/ui/adapters/DevicesSavedAdapter.java
package org.syncloud.android.ui.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import org.syncloud.android.Preferences;
import org.syncloud.android.R;
import org.syncloud.android.SyncloudApplication;
import org.syncloud.android.ui.DevicesSavedActivity;
import org.syncloud.android.core.platform.model.DomainModel;

public class DevicesSavedAdapter extends ArrayAdapter<DomainModel> {
    private final Preferences preferences;
    private DevicesSavedActivity activity;

    public DevicesSavedAdapter(DevicesSavedActivity activity) {
        super(activity, R.layout.layout_device_item);
        this.activity = activity;
        preferences = ((SyncloudApplication) activity.getApplication()).getPreferences();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        String mainDomain = preferences.getMainDomain();

        LayoutInflater inflater = activity.getLayoutInflater();
        View rowView = inflater.inflate(R.layout.layout_device_item, null);

        TextView txtBoldTitle = (TextView) rowView.findViewById(R.id.txt_bold_title);
        TextView txtAdditionalLine = (TextView) rowView.findViewById(R.id.txt_additional_line);

        final DomainModel domain = getItem(position);

        String fullDomainName = domain.userDomain()+"."+mainDomain;
        txtBoldTitle.setText(fullDomainName);

        txtAdditionalLine.setText(domain.id().title());

        return rowView;
    }

}

File: syncloud/src/main/java/org/syncloud/android/ui/adapters/DevicesSavedAdapter.kt
package org.syncloud.android.ui.adapters

import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import org.syncloud.android.R
import org.syncloud.android.core.platform.model.DomainModel
import org.syncloud.android.ui.DevicesSavedActivity

class DevicesSavedAdapter(private val activity: DevicesSavedActivity) : ArrayAdapter<DomainModel>(
    activity, R.layout.layout_device_item
) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = activity.layoutInflater
        val rowView = inflater.inflate(R.layout.layout_device_item, null)
        val txtBoldTitle = rowView.findViewById<TextView>(R.id.txt_bold_title)
        val txtAdditionalLine = rowView.findViewById<TextView>(R.id.txt_additional_line)
        val domain = getItem(position)!!
        val fullDomainName = domain.name
        txtBoldTitle.text = fullDomainName
        txtAdditionalLine.text = domain.title
        return rowView
    }
}